system/loggerd/tests/test_deleter.py

```diff
@@ -98,6 +98,50 @@ def test_no_delete_with_lock_file(self):
 
     self.assertTrue(f_path.exists(), "File deleted when locked")
 
+  def test_no_delete_preserved(self):
+    f_paths = []
+    for seg_num, preserve_attr in enumerate((None, deleter.PRESERVE_ATTR_VALUE, None)):
+      seg_dir = self.seg_format.format(seg_num)
+      f_paths.append(self.make_file_with_data(seg_dir, self.f_type, preserve_xattr=preserve_attr))
+
+    self.start_thread()
+
+    try:
+      # allow time for any files to be deleted
+      time.sleep(2)
+    finally:
+      self.join_thread()
+
+    self.join_thread()
+
+    self.assertTrue(f_paths[0].exists() and f_paths[1].exists(), "File deleted when preserved")
+    self.assertFalse(f_paths[2].exists(), "File not deleted")
+
+  def test_delete_oldest_preserved(self):
+    self.seg_dir = self.seg_format.format(self.seg_num)
+    f_path_old_preserved = self.make_file_with_data(self.seg_dir, self.f_type, preserve_xattr=deleter.PRESERVE_ATTR_VALUE)
+
+    f_paths_preserved = []
+    for _ in range(deleter.PRESERVE_COUNT):
+      # space out so they do not preserve each other
+      self.seg_num += 2
+      self.seg_dir = self.seg_format.format(self.seg_num)
+      f_paths_preserved.append(self.make_file_with_data(self.seg_dir, self.f_type, preserve_xattr=deleter.PRESERVE_ATTR_VALUE))
+
+    self.start_thread()
+
+    try:
+      with Timeout(2, "Timeout waiting for file to be deleted"):
+        while all([f_path.exists() for f_path in f_paths_preserved]):
+          time.sleep(0.01)
+    except TimeoutException:
+      pass
```

Comment:
what's the point of the timeout exception if it's just caught? just add to the while if you want that behavior